core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala

```diff
@@ -256,4 +259,38 @@ object TiUtil {
 
     mutableRow
   }
+
+  def getTiDBSnapshot(sparkSession: SparkSession): Option[TiTimestamp] = {
+    val staleReadTs =
+      sparkSession.conf.get(TiConfigConst.STALE_READ, TiConfigConst.DEFAULT_STALE_READ)
+    logger.info(s"${TiConfigConst.STALE_READ} = $staleReadTs")
+    if (staleReadTs.isEmpty) {
+      Option.empty
+    } else {
+      Some(parseTimestamp(staleReadTs))
+    }
+  }
+
+  private def parseTimestamp(str: String): TiTimestamp = {
+    if (!isValidTimestampMill(str)) {
+      throw new IllegalArgumentException(
+        "Invalid value of " + TiConfigConst.STALE_READ + ":" + str)
+    } else {
+      try {
+        val ts = java.lang.Long.parseLong(str)
+        new TiTimestamp(ts, 0L)
+      } catch {
+        case _: Throwable =>
+          throw new IllegalArgumentException(
+            "Unknown error to parse" + TiConfigConst.STALE_READ + ":" + str)
+      }
+    }
+  }
+
+  private def isValidTimestampMill(str: String): Boolean = {
+    if (StringUtils.isBlank(str) || !StringUtils.isNumeric(str)) {
+      return false
+    }
+    str.length() == 13;
```

Comment:
what's the point of judging the length

Reply:
The length of timestamp(ms) ts must be 13. check here to judge if the configuration is valid

Reply:
remove the limit of 13.
because it will be invalid after Nov 20 2286 at 17:46:39.999 UTC